### ISSUE TITLE
[BigQuery Dialect] Unparseable Pipe Syntax inside CTE

### ISSUE THREAD

**Issue by wesselvdg** [sqlfluff/sqlfluff#7153]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Whenever starting a CTE with the `FROM` Statement (when using the Pipe Syntax), the file becomes unparseable and returns 

```
!! Expected: "<Ref: 'SelectableGrammar'> to start sequence. Found <WordSegment: ([L:  8, P:  3]) 'FROM'>"
```

### Expected Behaviour

A small example containing both standard query and pipe syntax:

```sql
with foo as (SELECT 1 AS one_digit, 10 AS two_digit
    UNION ALL
    SELECT 2, 20
    UNION ALL
    SELECT 3, 30)

, bar AS (
  FROM foo
  )

SELECT one_digit, two_digit FROM bar
```

I expect it to parse correctly. All defaults settings are used:
`sqlfluff parse minimal_example.sql --dialect=bigquery` 

or with the .sqlfluff file from below:
`sqlfluff parse minimal_example.sql --ignore-local-config --config=.sqlfluff`

### Observed Behaviour

Observation is this error:

```
==== parsing violations ====
L:   8 | P:   3 |  PRS | Line 8, Position 3: Found unparsable section: 'FROM foo'
WARNING: Parsing errors found and dialect is set to 'bigquery'. Have you configured your dialect correctly?
```

Full trace:

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        with_compound_statement:
[L:  1, P:  1]      |            keyword:                                          'with'
[L:  1, P:  5]      |            whitespace:                                       ' '
[L:  1, P:  6]      |            common_table_expression:
[L:  1, P:  6]      |                naked_identifier:                             'foo'
[L:  1, P:  9]      |                whitespace:                                   ' '
[L:  1, P: 10]      |                keyword:                                      'as'
[L:  1, P: 12]      |                whitespace:                                   ' '
[L:  1, P: 13]      |                bracketed:
[L:  1, P: 13]      |                    start_bracket:                            '('
[L:  1, P: 14]      |                    [META] indent:
[L:  1, P: 14]      |                    set_expression:
[L:  1, P: 14]      |                        select_statement:
[L:  1, P: 14]      |                            select_clause:
[L:  1, P: 14]      |                                keyword:                      'SELECT'
[L:  1, P: 20]      |                                [META] indent:
[L:  1, P: 20]      |                                whitespace:                   ' '
[L:  1, P: 21]      |                                select_clause_element:
[L:  1, P: 21]      |                                    numeric_literal:          '1'
[L:  1, P: 22]      |                                    whitespace:               ' '
[L:  1, P: 23]      |                                    alias_expression:
[L:  1, P: 23]      |                                        [META] indent:
[L:  1, P: 23]      |                                        alias_operator:
[L:  1, P: 23]      |                                            keyword:          'AS'
[L:  1, P: 25]      |                                        whitespace:           ' '
[L:  1, P: 26]      |                                        naked_identifier:     'one_digit'
[L:  1, P: 35]      |                                        [META] dedent:
[L:  1, P: 35]      |                                comma:                        ','
[L:  1, P: 36]      |                                whitespace:                   ' '
[L:  1, P: 37]      |                                select_clause_element:
[L:  1, P: 37]      |                                    numeric_literal:          '10'
[L:  1, P: 39]      |                                    whitespace:               ' '
[L:  1, P: 40]      |                                    alias_expression:
[L:  1, P: 40]      |                                        [META] indent:
[L:  1, P: 40]      |                                        alias_operator:
[L:  1, P: 40]      |                                            keyword:          'AS'
[L:  1, P: 42]      |                                        whitespace:           ' '
[L:  1, P: 43]      |                                        naked_identifier:     'two_digit'
[L:  1, P: 52]      |                                        [META] dedent:
[L:  1, P: 52]      |                                [META] dedent:
[L:  1, P: 52]      |                        newline:                              '\n'
[L:  2, P:  1]      |                        whitespace:                           '    '
[L:  2, P:  5]      |                        set_operator:
[L:  2, P:  5]      |                            keyword:                          'UNION'
[L:  2, P: 10]      |                            whitespace:                       ' '
[L:  2, P: 11]      |                            keyword:                          'ALL'
[L:  2, P: 14]      |                        newline:                              '\n'
[L:  3, P:  1]      |                        whitespace:                           '    '
[L:  3, P:  5]      |                        select_statement:
[L:  3, P:  5]      |                            select_clause:
[L:  3, P:  5]      |                                keyword:                      'SELECT'
[L:  3, P: 11]      |                                [META] indent:
[L:  3, P: 11]      |                                whitespace:                   ' '
[L:  3, P: 12]      |                                select_clause_element:
[L:  3, P: 12]      |                                    numeric_literal:          '2'
[L:  3, P: 13]      |                                comma:                        ','
[L:  3, P: 14]      |                                whitespace:                   ' '
[L:  3, P: 15]      |                                select_clause_element:
[L:  3, P: 15]      |                                    numeric_literal:          '20'
[L:  3, P: 17]      |                                [META] dedent:
[L:  3, P: 17]      |                        newline:                              '\n'
[L:  4, P:  1]      |                        whitespace:                           '    '
[L:  4, P:  5]      |                        set_operator:
[L:  4, P:  5]      |                            keyword:                          'UNION'
[L:  4, P: 10]      |                            whitespace:                       ' '
[L:  4, P: 11]      |                            keyword:                          'ALL'
[L:  4, P: 14]      |                        newline:                              '\n'
[L:  5, P:  1]      |                        whitespace:                           '    '
[L:  5, P:  5]      |                        select_statement:
[L:  5, P:  5]      |                            select_clause:
[L:  5, P:  5]      |                                keyword:                      'SELECT'
[L:  5, P: 11]      |                                [META] indent:
[L:  5, P: 11]      |                                whitespace:                   ' '
[L:  5, P: 12]      |                                select_clause_element:
[L:  5, P: 12]      |                                    numeric_literal:          '3'
[L:  5, P: 13]      |                                comma:                        ','
[L:  5, P: 14]      |                                whitespace:                   ' '
[L:  5, P: 15]      |                                select_clause_element:
[L:  5, P: 15]      |                                    numeric_literal:          '30'
[L:  5, P: 17]      |                                [META] dedent:
[L:  5, P: 17]      |                    [META] dedent:
[L:  5, P: 17]      |                    end_bracket:                              ')'
[L:  5, P: 18]      |            newline:                                          '\n'
[L:  6, P:  1]      |            newline:                                          '\n'
[L:  7, P:  1]      |            comma:                                            ','
[L:  7, P:  2]      |            whitespace:                                       ' '
[L:  7, P:  3]      |            common_table_expression:
[L:  7, P:  3]      |                naked_identifier:                             'bar'
[L:  7, P:  6]      |                whitespace:                                   ' '
[L:  7, P:  7]      |                keyword:                                      'AS'
[L:  7, P:  9]      |                whitespace:                                   ' '
[L:  7, P: 10]      |                bracketed:
[L:  7, P: 10]      |                    start_bracket:                            '('
[L:  7, P: 11]      |                    [META] indent:
[L:  7, P: 11]      |                    newline:                                  '\n'
[L:  8, P:  1]      |                    whitespace:                               '  '
[L:  8, P:  3]      |                    unparsable:                               !! Expected: "<Ref: 'SelectableGrammar'> to start sequence. Found <WordSegment: ([L:  8, P:  3]) 'FROM'>"
[L:  8, P:  3]      |                        word:                                 'FROM'
[L:  8, P:  7]      |                        whitespace:                           ' '
[L:  8, P:  8]      |                        word:                                 'foo'
[L:  8, P: 11]      |                    newline:                                  '\n'
[L:  9, P:  1]      |                    whitespace:                               '  '
[L:  9, P:  3]      |                    [META] dedent:
[L:  9, P:  3]      |                    end_bracket:                              ')'
[L:  9, P:  4]      |            newline:                                          '\n'
[L: 10, P:  1]      |            newline:                                          '\n'
[L: 11, P:  1]      |            select_statement:
[L: 11, P:  1]      |                select_clause:
[L: 11, P:  1]      |                    keyword:                                  'SELECT'
[L: 11, P:  7]      |                    [META] indent:
[L: 11, P:  7]      |                    whitespace:                               ' '
[L: 11, P:  8]      |                    select_clause_element:
[L: 11, P:  8]      |                        column_reference:
[L: 11, P:  8]      |                            naked_identifier:                 'one_digit'
[L: 11, P: 17]      |                    comma:                                    ','
[L: 11, P: 18]      |                    whitespace:                               ' '
[L: 11, P: 19]      |                    select_clause_element:
[L: 11, P: 19]      |                        column_reference:
[L: 11, P: 19]      |                            naked_identifier:                 'two_digit'
[L: 11, P: 28]      |                    [META] dedent:
[L: 11, P: 28]      |                whitespace:                                   ' '
[L: 11, P: 29]      |                from_clause:
[L: 11, P: 29]      |                    keyword:                                  'FROM'
[L: 11, P: 33]      |                    whitespace:                               ' '
[L: 11, P: 34]      |                    from_expression:
[L: 11, P: 34]      |                        [META] indent:
[L: 11, P: 34]      |                        from_expression_element:
[L: 11, P: 34]      |                            table_expression:
[L: 11, P: 34]      |                                table_reference:
[L: 11, P: 34]      |                                    naked_identifier:         'bar'
[L: 11, P: 37]      |                        [META] dedent:
[L: 11, P: 37]      |    [META] end_of_file:

```

### How to reproduce

You can run the minimal example with the command
 `sqlfluff parse minimal_example.sql --dialect=bigquery` 

or use the .sqlfluff file below:
`sqlfluff parse minimal_example.sql --ignore-local-config --config=.sqlfluff`

[minimal_example.sql](https://github.com/user-attachments/files/22656219/minimal_example.sql)

### Dialect

BigQuery

### Version

```
sqlfluff --version  
> sqlfluff, version 3.4.2

```

### Configuration

.sqlfluff:

```
[sqlfluff:core]
dialect = bigquery
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by wesselvdg** [comment]
Creating a different CTEDefinitionSegment seemed to work for me, however I doubt this would be the best solution.

```python
class CTEDefinitionSegment(BaseSegment):
    """A CTE Definition from a WITH statement.

    BigQuery allows FROM clauses directly in CTEs without requiring SELECT statements.
    This extends the ANSI definition to support this pipe syntax.

    Examples:
        WITH cte AS (SELECT * FROM table)  -- Standard syntax
        WITH cte AS (FROM table)           -- Pipe syntax
    """
    type = "common_table_expression"

    match_grammar: Matchable = Sequence(
        Ref("SingleIdentifierGrammar"),
        Ref("CTEColumnList", optional=True),
        Ref.keyword("AS", optional=True),
        Bracketed(
            # Allow FROM clause directly in CTE, or standard SelectableGrammar
            OneOf(
                Sequence(
                    "FROM",
                    Ref("TableReferenceSegment"),
                    AnyNumberOf(Ref("PipeOperatorClauseSegment")),
                ),
                Ref("SelectableGrammar"),
            ),
        ),
    )

    def get_identifier(self) -> IdentifierSegment:
        """Get the identifier of this CTE using ANSI's implementation."""
        return ansi.CTEDefinitionSegment.get_identifier(self)
```

**Comment by keraion** [comment]
I think you're on the right track here. 

You can replace your added sequence with `Ref("PipeStatementSegment")` to simplify it a bit. Also, you can inherit from `ansi.CTEDefinitionSegment` so you don't need to implement the `type` or `get_identifier`.

If you'd like to pull together a PR that includes your example as a test, I'm happy to review.

**Comment by annebelleo** [comment]
@wesselvdg I wanted to take a stab at this, hope you don't mind that I opened a PR!
